### ISSUE TITLE
Right clicking a shulker in action bar still places the shulker

### ISSUE THREAD

**Issue by MeeniMc** [kyrptonaught/quickshulker#16]
## TL;DR: 

in some cases, the in-hand shulker box is still placed, despite the fact that a shulker item was clicked in the inventory *(as opposed to right clicking a target block in-game).*

## Cases analysis

1. on-hand shulker, right click in-game with no target block (e.g., looking up the sky): the on-hand shulker is quick-opened (correct behavior).
2. on-hand shulker, right click in-game with a target block (e.g., looking at a grass block): the on-hand shulker is placed (correct behavior).
3. on-hand shulker, targeting a grass block, **right-click in-intentory** on the on-hand shulker: the shulker is placed **NOT CORRECT** *(this is not consistent, it may take a few tries (2 or 3) to see it happen).*
4. on-hand shulker, targeting a grass block, **right-click in-inventory on another** shulker: the on-hand shulker is placed **NOT CORRECT**, the clicked shulker is quick opened (correct). *(this may take a few more tries to see it).*

v 1.1.11-1.16.4, no other mods

**Comment by kyrptonaught** [comment]
I haven't been able to fix this one, but at least its only an annoyance in survival with the dupe-ability contained to creative only. I think it's due to the fact that when you open a new screen there is a few ticks where there is no screen open, thus allowing your right click that most likely is still being activated, to trigger a block place.

**Comment by MeeniMc** [comment]
I was about to report another issue but I remembered about this one, and I think it's the same underlying problem: 

Sometimes, when you are opening/closing boxes in your inventory, but there is a chest (or some other inventory block) "behind" your inventory, the click goes 'through' the inventory interface, and opens the targeted chest, rather than quick opening/closing the hovered shulker box. This behavior appears to confirm your understanding of the bug scenario.
